src/query/ast/src/ast/format/syntax/mod.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod common;
+mod ddl;
+mod dml;
+mod expr;
+mod query;
+
+use common_exception::Result;
+use ddl::*;
+use dml::*;
+use pretty::RcDoc;
+use query::*;
+
+use crate::ast::Statement;
+
+pub fn format_statement_syntax(stmt: Statement, max_width: usize) -> Result<String> {
+    let mut bs = Vec::new();
+    statement_doc(stmt).render(max_width, &mut bs)?;
+    Ok(String::from_utf8(bs)?)
+}
+
+fn statement_doc(stmt: Statement) -> RcDoc {
```

Comment:
The naming is quite confusing to me.

Maybe we can add a function `pretty_statement(stmt: Statement, max_width: usize) -> Result<String>`?